pkg/platform/model/buildplanner.go

```diff
@@ -160,56 +169,69 @@ func (bp *BuildPlanner) FetchBuildResult(commitID strfmt.UUID, owner, project st
 
 	// Filter the build terminals to only include the current platform
 	var filteredTerminals []*bpModel.NamedTarget
-	for _, t := range resp.Project.Commit.Build.Terminals {
+	for _, t := range build.Terminals {
 		if platformID.String() == strings.TrimPrefix(t.Tag, "platform:") {
 			filteredTerminals = append(filteredTerminals, t)
 		}
 	}
-	resp.Project.Commit.Build.Terminals = filteredTerminals
+	build.Terminals = filteredTerminals
 
 	buildEngine := Alternative
-	for _, s := range resp.Project.Commit.Build.Sources {
+	for _, s := range build.Sources {
 		if s.Namespace == "builder" && s.Name == "camel" {
 			buildEngine = Camel
 			break
 		}
 	}
 
+	id, err := resp.CommitID()
+	if err != nil {
+		return nil, errs.Wrap(err, "Response does not contain commitID")
+	}
+
 	res := BuildResult{
 		BuildEngine: buildEngine,
-		Build:       resp.Project.Commit.Build,
-		BuildReady:  resp.Project.Commit.Build.Status == bpModel.Ready,
-		CommitID:    strfmt.UUID(resp.Project.Commit.CommitID),
+		Build:       build,
+		BuildReady:  build.Status == bpModel.Completed,
+		CommitID:    id,
 	}
 
 	// We want to extract the recipe ID from the BuildLogIDs.
 	// We do this because if the build is in progress we will need to reciepe ID to
 	// initialize the build log streamer.
-	// For camel builds the ID type will not be BuildLogRecipeID but this is okay
-	// because the state tool does not display in progress information for camel builds.
-	for _, id := range resp.Project.Commit.Build.BuildLogIDs {
-		if id.Type == bpModel.BuildLogRecipeID {
-			if res.RecipeID != "" {
-				return nil, errs.Wrap(err, "Build plan contains multiple recipe IDs")
-			}
-			res.RecipeID = strfmt.UUID(id.ID)
+	// This information will only be populated if the build is an alternate build.
+	// This is specified in the build planner queries.
+	for _, id := range build.BuildLogIDs {
+		if res.RecipeID != "" {
+			return nil, errs.Wrap(err, "Build plan contains multiple recipe IDs")
 		}
+		res.RecipeID = strfmt.UUID(id.ID)
```

Comment:
```suggestion
		res.RecipeID = strfmt.UUID(id.ID)
		break
```

Or do we expect multiple matches and only want the last one?

Reply:
No, you're right. There should only be one.